### ISSUE TITLE
An internal error has occurred. Please report this as a bug.

### ISSUE THREAD

**Issue by goerch** [agda/agda#1877]
Location of the error: src/full/Agda/TypeChecking/Monad/MetaVars.hs:81
[issue.agda.txt](https://github.com/agda/agda/files/147808/issue.agda.txt)

**Comment by goerch** [comment]
Tested with version 2.5.0. Worked in earlier versions.

**Comment by goerch** [comment]
Sorry, wrong button.

**Comment by UlfNorell** [comment]
Here's the test case for convenience

``` agda
module _ where

open import Level

open import Relation.Binary.PropositionalEquality

record Equivalence {c₁} (A : Set c₁) (c₂ : Level) : Set (c₁ ⊔ suc c₂) where
  constructor mk
  infix  4 _≈_    
  field
    _≈_ : A → A → Set c₂
    reflexive : {x : A} → x ≈ x 
    symmetric :  {x y : A} (e₁ : x ≈ y) → y ≈ x 
    transitive : {x y z : A} (e₁ : x ≈ y) (e₂ : y ≈ z) → x ≈ z

open Equivalence {{...}} using (_≈_)

instance
  propositionalEquivalence : {c : Level} {A : Set c} → Equivalence A c
  propositionalEquivalence = record { _≈_ = _≡_ ; reflexive = refl ; symmetric = sym ; transitive = trans }

record Monoid (c : Level) : Set (suc c) where
  constructor mk
  infixl 7 _∙_
  field
    set : Set c
    ε : set
    _∙_ : set → set → set

    id-left : {x : set} → ε ∙ x ≈ x
    id-right : {x : set} → x ∙ ε ≈ x 
    assoc : {x y z : set} → x ∙ (y ∙ z) ≈ (x ∙ y) ∙ z

open Monoid {{...}}

record MonoidMorphism {c : Level} (A B : Monoid c)  : Set (suc c) where
  constructor mk
  field
    apply : set {{A}} → set {{B}}
    apply-respects-ε : apply (ε {{A}}) ≈ ε {{B}}
    apply-respects-∙ : ∀ {x y} → apply (x ∙ y) ≈ apply x ∙ apply y
```

**Comment by andreasabel** [comment]
This checks fine on maint-2.4.

**Comment by asr** [comment]
With Agda 2.5.1 RC 1 I got the error:

```
/tmp/issue.agda:41,43-44
No instance of type Monoid c was found in scope.
when checking that x y are valid arguments to a function of type
{c = c₁ : Level} {{r : Monoid c₁}} →
Monoid.set r → Monoid.set r → Monoid.set r
```

**Comment by UlfNorell** [comment]
Bug introduced by fix of #1865 (5aca8aa).
